### ISSUE TITLE
Ffuf causes autorecon to hang endlessly

### ISSUE THREAD

**Issue by 5a3ee** [Tib3rius/AutoRecon#85]
All ffuf processes on my system are now hanging. I can only get autorecon to continue and detect ffuf as finished by cat'ing /proc/pidofffuf/fd/2, which raises the error below in Autorecon:
```
Task exception was never retrieved
future: <Task finished name='Task-173' coro=<read_stream() done, defined at /home/user/.local/pipx/venvs/autorecon/lib/python3.9/site-packages/autorecon/autorecon.py:221> exception=ValueError('Separator is not found, and chunk exceed the limit')>
Traceback (most recent call last):
  File "/usr/lib/python3.9/asyncio/streams.py", line 540, in readline
    line = await self.readuntil(sep)
  File "/usr/lib/python3.9/asyncio/streams.py", line 618, in readuntil
    raise exceptions.LimitOverrunError(
asyncio.exceptions.LimitOverrunError: Separator is not found, and chunk exceed the limit

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/user/.local/pipx/venvs/autorecon/lib/python3.9/site-packages/autorecon/autorecon.py", line 224, in read_stream
    line = await stream.readline()
  File "/usr/lib/python3.9/asyncio/streams.py", line 549, in readline
    raise ValueError(e.args[0])
ValueError: Separator is not found, and chunk exceed the limit
```

**Comment by zngt2** [comment]
Same here but don't have the rout that you mentioned
(/proc/pidofffuf/fd/2)

**Comment by 5a3ee** [comment]
@zngt2 You have to change pidofffuf with the PID of ffuf. Although this one liner would do the same:
```sh
cat /proc/$(pidof fuff)/fd/2
```

**Comment by Coen-Schuijt** [comment]
It's do to an overflow on the limit size.

Make the following change on line 277 in _/home/kali/.local/pipx/venvs/autorecon/lib/python3.9/site-packages/autorecon/autorecon.py_ (or wherever your autorecon resides):

`process = await asyncio.create_subprocess_shell(cmd, stdout=asyncio.subprocess.PIPE, stderr=asyncio.subprocess.PIPE, executable='/bin/bash')`

To
`process = await asyncio.create_subprocess_shell(cmd, stdout=asyncio.subprocess.PIPE, stderr=asyncio.subprocess.PIPE, limit=1024 * 256, executable='/bin/bash')`

**Comment by sudoplswork** [comment]
@Coen-Schuijt 

I noticed that setting the limit to 256kb was insufficient in preventing errors for me, I further increased the limit to `limit=1024 * 512`which resolved my issue. 

If anyone else is having this issue I suggest they try the same.

**Comment by w33vils** [comment]
Did anyone find a fix to the hanging? Appreciated

**Comment by ZevSwe** [comment]
Just confirming the find. It seems it sometimes works and sometimes hangs. I guess just changing ffuf to http.manual on line 128 in config/service-scans-default.toml would do the trick for now.

**Comment by Tib3rius** [comment]
Fixed in 4a97eb9.
